optimum/onnxruntime/base.py

```diff
@@ -227,6 +235,14 @@ def prepare_inputs_for_merged(
                 past_key_values = tuple(
                     key_or_value for _ in range(len(self.key_value_input_names) // 2) for key_or_value in [key, value]
                 )
+            elif self.parent_model.config.model_type in MULTI_QUERY_ATTN_MODELS:
```

Comment:
This kind of controlflow that we have more and more (for bloom, now models that use multi-query attention) and that is run in the forward makes me think we should somehow subclass, and have `ORTBloomForCausalLM`, `ORTModelForCausalLMWithMQA`.

The solution in https://github.com/huggingface/optimum/pull/1152 is very ugly, but somehow I still think we should be more modular (not in this PR obviously!)

What do you think?

Reply:
Yes, I also thought of that. It would be better to have separate decoder modeling other than adding these control flows. I just wanted to be align with the case of bloom. We can target a refactoring for that!